### PR TITLE
Add feedback submitted translator event

### PR DESCRIPTION
**Pull request by garemoko**
Adds a feedback submitted event that catches the event added in https://github.com/LearningLocker/Moodle-Log-Expander/pull/26 and feeds into the existing attempt completed emitter event. This works, in that it sends the desired statement to the LRS. 

I still need to do:
- [x] tests
- [x] documentation
- [x] feedback question submitted

FeedbackQuestionSubmitted will extend FeedbackSubmitted to take advantage of the parseFeedback function. This does mean that this function gets called twice; once for each translator event.

### CODE REVIEW

**Comment by garemoko** [comment]
To fix: both feedback submitted and question submitted extensions include only 1 question and all responses. The feedback should include all questions; the question should include only 1 response.

**Comment by garemoko** [comment]
@jlowe64 @ryansmith94 this is now done and ready for review. 

Remember that the Log Expander PR for feedback was merged a few weeks ago; also, there will be no Emitter PR as this uses the existing attempt completed and question answered recipes. Therefore once this merges, tracking of the Feedback module (https://github.com/jlowe64/moodle-logstore_xapi/issues/51) is done.

File: src/Events/FeedbackSubmitted.php [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+            return array();
+        }
+
+        // Strip out the junk.
+        $presentation = str_replace('r>>>>>', '', $presentation);
+        $presentation = trim(preg_replace('/\s+/', ' ', $presentation));
+        $presentation = strip_tags($presentation);
+
+        $options = explode('|', $presentation);
+        $return = array((object)[
```
Comment (ryansmith94) [inline src/Events/FeedbackSubmitted.php]: Use `[...]` syntax rather than `array` function. This comment also applies elsewhere (such as line 133).

**Comment by ryansmith94** [comment]
Besides my one comment, this looks fine to me. Might need some integration tests for this in the [test directory of the plugin](https://github.com/jlowe64/moodle-logstore_xapi/tree/master/tests) though.

**Comment by garemoko** [comment]
@ryansmith94 done. I don't see any event specific tests in the test folder of the plugin.

**Comment by ryansmith94** [comment]
@garemoko probably want an integration test for "\mod_feedback\event\response_submitted" event similar to the [AttemptStartedTest](https://github.com/jlowe64/moodle-logstore_xapi/blob/master/tests/AttemptStartedTest.php).

**Comment by garemoko** [comment]
@ryansmith94 sorry, I misunderstood what you meant by integration tests. There are tests in this PR:

https://github.com/LearningLocker/Moodle-xAPI-Translator/pull/39/files#diff-264ca5f4a3860b46f343b0d4bb76020fR1

https://github.com/LearningLocker/Moodle-xAPI-Translator/pull/39/files#diff-ef764371d5eb08311bfb2837b3663b4cR1

**Comment by ryansmith94** [comment]
I can see the tests you've added to this repo in this PR. However, the plugin also has it's own integration tests to ensure that the Expander, Translator, and Emitter integrate correctly.

**Comment by garemoko** [comment]
OK. I'm on the right page now. For some reason those tests aren't in my copy of the plugin, which confused me.

**Comment by ryansmith94** [comment]
Ah right no worries.

**Comment by jlowe64** [comment]
@garemoko I'm integrating this into the plugin repo and I've run into a bit of an error and need some clarification:

Turn to line 45 and 46 of FeedbackSubmitted in this repo:
```
foreach ($opts['questions'] as $item => $question) {
            // Find the response to the current question
            $currentResponse = null;
            foreach ($opts['attempt']->responses as $responseId => $response) {
                if ($response->item == $item) {
                    $currentResponse = $response;
                }
            }
```

When you say ` if ($response->item == $item) {` do you actually mean ` if ($responseId == $item) {` or do you mean something else? There is no $item variable defined within `$opts['attempt']->responses`.

**Comment by garemoko** [comment]
I'd have to check the code, but $item is defined in scope from the sample above - it's the index of the array of questions. I'll look in more detail when I'm at a computer.  What's the error you are hitting?

**Comment by garemoko** [comment]
Oh sorry I misread the question. I'll look properly later.

**Comment by jlowe64** [comment]
Thanks @garemoko

**Comment by garemoko** [comment]
https://github.com/moodle/moodle/blob/master/mod/feedback/db/install.xml

**Comment by garemoko** [comment]
https://github.com/LearningLocker/Moodle-Log-Expander/blob/master/src/Repository.php

**Comment by garemoko** [comment]
OK,  I actually managed to figure this out on my phone. I think this another gap in the test repository.  There is an item field in the feedback_values table, which is where the responses object is ultimately  pulling from.

**Comment by jlowe64** [comment]
Ahhh, I will take a look and see if I can pull a proper example from the database and populate this for you if you want? Would only be a few minutes work.

**Comment by garemoko** [comment]
Yes please!

**Comment by jlowe64** [comment]
@garemoko thanks for helping!

**Pull request merged**